repo2docker/app.py

```diff
@@ -629,9 +630,12 @@ def wait_for_container(self, container):
         Displaying logs while it's running
         """
 
+        last_timestamp = None
         try:
-            for line in container.logs(stream=True):
-                self.log.info(line.decode("utf-8"), extra=dict(phase="running"))
+            for line in container.logs(stream=True, timestamps=True):
```

Comment:
This will need an addition to
https://github.com/jupyterhub/repo2docker/blob/eb7520901913d61e7745d136ce127142d5fa8506/repo2docker/engine.py#L24-L36
https://github.com/jupyterhub/repo2docker/blob/eb7520901913d61e7745d136ce127142d5fa8506/repo2docker/docker.py#L16-L17

and a spec of what it should return. Logging was one of the big problems I had in https://github.com/jupyterhub/repo2docker/pull/848, see the comment about `Build logs` in the PR description.

Reply:
Thanks! Exposed `timestamps` and `since` which are required to avoid duplication of output.

Reply:
Docker `logs(since)` has a quirk that it only seems to accept integer timestamps. Since we now have a wrapper layer, what do you think we should accept here? A datetime object? An integer timestamp like docker itself? A timestamp string as prefixed when `timestamps=True`?

Reply:
Not sure.... They all seem reasonable, so go with whatever you want.

Reply:
Went with symmetry, so our `since` accepts timestamps exactly as they appear, and moved the cast-to-int quirk inside the Docker implementation